From 1537d131e59e4fb43001299cfbd747c521fa1888 Mon Sep 17 00:00:00 2001
From: Mike McQuaid <mike.mcquaid@kdab.com>
Date: Thu, 21 Jul 2011 10:35:01 -0400
Subject: [PATCH] Fix compilation under OSX 10.7 or using llvm-gcc.

Use correct error codes instead of type errors.
Thanks to Dylan Luke <lukes.dylan@gmail.com> for this patch.
---
 src/gui/kernel/qcocoasharedwindowmethods_mac_p.h |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/gui/kernel/qcocoasharedwindowmethods_mac_p.h b/src/gui/kernel/qcocoasharedwindowmethods_mac_p.h
index 9c110fd..94974fc 100644
--- src/gui/kernel/qcocoasharedwindowmethods_mac_p.h
+++ src/gui/kernel/qcocoasharedwindowmethods_mac_p.h
@@ -309,7 +309,7 @@ QT_END_NAMESPACE
 
     QWidget *target = [self dragTargetHitTest:sender];
     if (!target)
-        return [super draggingEntered:sender];
+        return NSDragOperationNone;
     if (target->testAttribute(Qt::WA_DropSiteRegistered) == false)
         return NSDragOperationNone;
 
@@ -321,7 +321,7 @@ QT_END_NAMESPACE
 {
     QWidget *target = [self dragTargetHitTest:sender];
     if (!target)
-        return [super draggingUpdated:sender];
+        return NSDragOperationNone;
 
     if (target == *currentDragTarget()) {
         // The drag continues to move over the widget that we have sendt
@@ -345,7 +345,7 @@ QT_END_NAMESPACE
 {
     QWidget *target = [self dragTargetHitTest:sender];
     if (!target)
-        return [super draggingExited:sender];
+        return;
 
     if (*currentDragTarget()) {
         [reinterpret_cast<NSView *>((*currentDragTarget())->winId()) draggingExited:sender];
@@ -357,7 +357,7 @@ QT_END_NAMESPACE
 {
     QWidget *target = [self dragTargetHitTest:sender];
     if (!target)
-        return [super performDragOperation:sender];
+        return NO;
 
     BOOL dropResult = NO;
     if (*currentDragTarget()) {
-- 
1.6.1

